### PR TITLE
backend/feat/split_settlement

### PR DESCRIPTION
**Pull request by 0utkarsh**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Payment.hs [hunk 1]
```diff
@@ -97,7 +98,8 @@ createOrder (driverId, merchantId, opCityId) invoiceId = do
     CQSC.findSubscriptionConfigsByMerchantOpCityIdAndServiceName opCityId serviceName
       >>= fromMaybeM (NoSubscriptionConfigForService opCityId.getId $ show serviceName)
   let paymentServiceName = subscriptionConfig.paymentServiceName
-  (createOrderResp, _) <- SPayment.createOrder (driverId, merchantId, opCityId) paymentServiceName (catMaybes driverFees, []) Nothing INV.MANUAL_INVOICE (getIdAndShortId <$> listToMaybe invoices) Nothing
+  vendorFees <- mapM QVF.findByDriverFeeId (map Domain.Types.DriverFee.id (catMaybes driverFees))
```
Comment (aayushagarwal2001) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Payment.hs:101]: bro, can we have a flag in subsconfig for this query :-
`vendorFees <- mapM QVF.findByDriverFeeId (map Domain.Types.DriverFee.id (catMaybes driverFees))`
like :- 
`if isEnabledVendorSplit then mapM QVF.findByDriverFeeId (map Domain.Types.DriverFee.id (catMaybes driverFees)) else []`

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/VendorSplitDetailsExtra.hs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+-- Extra code goes here --
+
+findAllByAreaCityAndCategory ::
+  (EsqDBFlow m r, MonadFlow m, CacheFlow m r) =>
+  Area ->
+  Id MerchantOperatingCity ->
+  VehicleCategory ->
+  m [VendorSplitDetails]
+findAllByAreaCityAndCategory area merchantOperatingCityId vehicleCategory =
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/VendorSplitDetailsExtra.hs:28]: Any challenges in moving this Query to DSL ?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/VendorFeeExtra.hs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import Kernel.Types.Price
+import Kernel.Utils.Common (CacheFlow, EsqDBFlow, MonadFlow, fromMaybeM, getCurrentTime)
+import qualified Sequelize as Se
+import qualified Storage.Beam.VendorFee as Beam
+import Storage.Queries.OrphanInstances.VendorFee
+
+-- Extra code goes here --
+
+findAllByDriverFeeId :: (EsqDBFlow m r, MonadFlow m, CacheFlow m r) => Id DriverFee -> m [VendorFee]
+findAllByDriverFeeId driverFeeId = findAllWithKV [Se.Is Beam.driverFeeId $ Se.Eq driverFeeId.getId]
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/VendorFeeExtra.hs:22]: Add a DB Index on DriverFeeId.

File: Backend/dev/migrations/dynamic-offer-driver-app/0546-vendor-split-details.sql [hunk 4]
```diff
@@ -0,0 +1,270 @@
+UPDATE atlas_driver_offer_bpp.subscription_config SET is_vendor_split_enabled = true where merchant_operating_city_id in (select id from atlas_driver_offer_bpp.merchant_operating_city where city in ('Kolkata', 'Asansol', 'Durgapur', 'Siliguri'));
+
+-- Kolkata Open Market --
+-- 2 Rs. for 1233333 (any variant)
```
Comment (khuzema786) [inline Backend/dev/migrations/dynamic-offer-driver-app/0546-vendor-split-details.sql:4]: This Vendor Id can be UUID ?

Comment (0utkarsh) [reply Backend/dev/migrations/dynamic-offer-driver-app/0546-vendor-split-details.sql:4]: It is Juspay sub account id

**Review by khuzema786 — APPROVED** [review]

**Pull request merged**